mobile/src/main/java/org/openhab/habdroid/util/CacheManager.kt

```diff
@@ -38,19 +37,37 @@ class CacheManager private constructor(appContext: Context) {
         // int in its constructor.
         val maxMemory = (Runtime.getRuntime().maxMemory() / 1024).toInt()
         // Use up to 1/8 of the available VM memory for the bitmap cache
-        bitmapCache = object : LruCache<HttpUrl, Bitmap>(maxMemory / 8) {
+        iconBitmapCache = object : LruCache<HttpUrl, Bitmap>(maxMemory / 8) {
+            override fun sizeOf(key: HttpUrl, value: Bitmap): Int {
+                return value.byteCount / 1024
+            }
+        }
+        temporaryBitmapCache = object : LruCache<HttpUrl, Bitmap>(maxMemory / 8) {
             override fun sizeOf(key: HttpUrl, value: Bitmap): Int {
                 return value.byteCount / 1024
             }
         }
     }
 
     fun getCachedBitmap(url: HttpUrl): Bitmap? {
-        return bitmapCache.get(url)
+        return if (isIconUrl(url)) {
+            iconBitmapCache.get(url)
+        } else {
+            temporaryBitmapCache.get(url)
+        }
     }
 
     fun cacheBitmap(url: HttpUrl, bitmap: Bitmap) {
-        bitmapCache.put(url, bitmap)
+        if (isIconUrl(url)) {
+            iconBitmapCache.put(url, bitmap)
+        } else {
+            temporaryBitmapCache.put(url, bitmap)
+        }
+    }
+
+    private fun isIconUrl(url: HttpUrl): Boolean {
```

Comment:
Maybe make this method `private fun targetCache(url: HttpUrl): LruCache<Bitmap>` and return the appropriate one here? Doing so would simplify the methods above.

Reply:
Good idea!